src/util/SunPosition.js

```diff
@@ -0,0 +1,173 @@
+             * @param {{declination: Number, rightAscension: Number}} celestialLocation
+             * @param {Date} date
+             * @throws {ArgumentError} if celestialLocation or julianDate are missing
+             * @return {{latitude: Number, longitude: Number}} the geographic location
+             */
+            celestialToGeographic: function (celestialLocation, date) {
+                if (!celestialLocation) {
+                    throw new ArgumentError(
+                        Logger.logMessage(Logger.LEVEL_SEVERE, "SunPosition", "celestialToGeographic",
+                            "missingCelestialLocation"));
+                }
+                if (date instanceof Date === false) {
+                    throw new ArgumentError(
+                        Logger.logMessage(Logger.LEVEL_SEVERE, "SunPosition", "celestialToGeographic", "missingDate"));
+                }
+
+                var julianDate = this.computeJulianDate(date);
+
+                //number of days (positive or negative) since Greenwich noon, Terrestrial Time, on 1 January 2000 (J2000.0)
+                var numDays = julianDate - 2451545;
+
+                //Greenwich Mean Sidereal Time
+                var GMST = this.normalizeAngle(280.46061837 + 360.98564736629 * numDays);
+
+                //Greenwich Hour Angle
+                var GHA = this.normalizeAngle(GMST - celestialLocation.rightAscension);
+
+                var longitude = Angle.normalizedDegreesLongitude(-GHA);
+
+                return {
+                    latitude: celestialLocation.declination,
+                    longitude: longitude
+                };
+            },
+
+            /**
+             * Computes the julian date from a javascript date object
+             * @param {Date} date
+             * @throws {ArgumentError} if the date is missing
+             * @return {Number} the julian date
+             */
+            computeJulianDate: function (date) {
+                if (date instanceof Date === false) {
+                    throw new ArgumentError(
+                        Logger.logMessage(Logger.LEVEL_SEVERE, "SunPosition", "computeJulianDate", "missingDate"));
+                }
+
+                var year = date.getUTCFullYear();
+                var month = date.getUTCMonth() + 1;
+                var day = date.getUTCDate();
+                var hour = date.getUTCHours();
+                var minute = date.getUTCMinutes();
+                var second = date.getUTCSeconds();
+
+                var dayFraction = (hour + minute / 60 + second / 3600) / 24;
+
+                if (month <= 2) {
+                    year -= 1;
+                    month += 12;
+                }
+
+                var A = Math.floor(year / 100);
+                var B = 2 - A + Math.floor(A / 4);
+                var JD0h = Math.floor(365.25 * (year + 4716)) + Math.floor(30.6001 * (month + 1)) + day + B - 1524.5;
+
+                return JD0h + dayFraction;
+            },
+
```

Comment:
@strikerM, This function looks appropriate for WWMath. Would you please move it there? Also, if you'd compare it to Android's WWMath.normalizeAngle we need to ensure that the two are equivalent.